apps/api/src/deployment/controllers/deployment/deployment.controller.ts

```diff
@@ -86,4 +88,31 @@ export class DeploymentController {
 
     return { data: result };
   }
+
+  @Protected([{ action: "sign", subject: "UserWallet" }])
+  async list({ skip, limit }: { skip?: number; limit?: number }): Promise<z.infer<typeof ListDeploymentsResponseSchema>> {
+    const { currentUser, ability } = this.authService;
+
+    const userWallet = await this.userWalletRepository.accessibleBy(ability, "sign").findOneByUserId(currentUser.id);
+    assert(userWallet, 404, "UserWallet Not Found");
+    assert(
+      (skip === undefined && limit === undefined) || (skip !== undefined && limit !== undefined),
```

Comment:
suggestion(blocking): zod schema can be used for this

Reply:
good way to approach pagination is to enforce some reasonable limit. like `1000` by default.